### ISSUE TITLE
Failed to upload ipynb file if it contains many images

### ISSUE THREAD

**Issue by shaolifu** [jupyterlab/jupyterlab#5038]
When upload some ipynb file which contains many image outputs, the browser will throw an exception: 

- JSON.parse: unterminated string at line 199 column 24964 of the JSON data (firefox)
- Unexpected end of JSON input (chrome)

If clean all outputs, it can be uploaded successfully.

**It's ok in the classic notebook mode.**

The following screenshot just for reference.

Thanks.

```Python
%matplotlib inline
import matplotlib.pyplot as plt
plt.rcParams["figure.figsize"] = [20, 9]

import numpy as np

np.random.seed(1)

plt.plot(np.random.randn(1000))

# many same cells, maybe 30 is enough
```

![image](https://user-images.githubusercontent.com/16026142/43620734-c63c3f22-9706-11e8-8eb3-75e227b689dd.png)

![image](https://user-images.githubusercontent.com/16026142/43620890-682b343c-9707-11e8-87a4-e24f4e6fcd54.png)

**Comment by jasongrout** [comment]
So you have a large such ipynb file and you *upload* it? By dragging into the file browser, or clicking the upload button? How big is your file?

**Comment by shaolifu** [comment]
Thanks!

I use the sample code to generate an ipynb file. The size may be 3 or 4 Mb. not too big.

But in the file, some tags' value are very long, such as follows

```Json
{
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<matplotlib.lines.Line2D at 0x7f1f42880a90>]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
          "image/png": "iVBORw0KGgoAAAA.........  ",      //  very long
      "text/plain": [
       "<Figure size 1440x648 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.plot(np.random.randn(1000))"
   ]
  }

``` 

If I  drag it to the file browser, no messages return but the file is not uploaded sucessfully. 

Using the upload button will show me the error message.

Maybe you can use the sample code to generate an ipynb file. I think it can be reproduced.

**Comment by shaolifu** [comment]
I attach the sample file, for your convenience.

[test.zip](https://github.com/jupyterlab/jupyterlab/files/2256568/test.zip)

**Comment by jasongrout** [comment]
CC @saulshanabrook, who worked on the upload code recently...

**Comment by blink1073** [comment]
Fixed by #5135
